x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/support/SecurityIndexManager.java

```diff
@@ -267,6 +270,21 @@ private static boolean isCreatedOnLatestVersion(IndexMetadata indexMetadata) {
         return indexVersionCreated != null && indexVersionCreated.onOrAfter(IndexVersion.current());
     }
 
+    /**
+     * Check to see if there are any file based role mappings and if they have been loaded into cluster state. If the
+     * {@link ReservedStateMetadata} file_settings namespace contains role mapping names, it means that there should be the same number of
+     * cluster state role mappings available. If they're not available yet using {@code RoleMappingMetadata.getFromClusterState()}, they
+     * have not yet been synchronized.
+     */
+    private static boolean isReadyForRoleMappingCleanupMigration(ClusterState clusterState) {
```

Comment:
Great idea! I actually started with an enum, then when we had the cluster mappings available in the security index manager state, I removed it (since we could just check size == 0 and return in the migration). Now we've come full circle and I've added it back :) 

I've also added a status `DONE` so we don't have to do these calculations on every security index manager change forever, which is really nice.